classes/items/itembase.cpp

```diff
@@ -18,44 +18,58 @@ void Item::generate()
 
 
 //Constructors / Destructors
-Item::Item()
-{
+Item::Item(name, type, tier, value, level);
+
     std::string name;
-    unsigned type;
-    unsigned value;
+    unsigned int type;
+    unsigned int value;
+    unsigned int tier;
+    unsigned int level;
+
+    this->name = name; //assigns the given value of name
+    this->type = type;  //assigns the given value of type
+    this->value = value; //assigns the given value of value
+    this->tier = tier; //assigns the given value of tier.
+    this->level = level;
```

Comment:
Change ; to { on the next line.

Then it should be fixed here to make it compile.